### PR TITLE
Fixes #10142: Extend variable_dict_from_file to allow reading yaml and csv files

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/10142

### CODE REVIEW

**Comment by amousset** [comment]
Not tested, do not merge for now.

File: tree/30_generic_methods/variable_dict_from_file_type.cf [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#####################################################################################
+# Copyright 2015 Normation SAS
```
Comment (peckpeck) [inline tree/30_generic_methods/variable_dict_from_file_type.cf]: 2017

File: tree/30_generic_methods/variable_dict_from_file_type.cf [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+bundle agent variable_dict_from_file_type(variable_prefix, variable_name, file_name, file_type)
+{
+  vars:
+      "old_class_prefix"  string => canonify("variable_dict_from_file_type_${variable_name}");
+      "promisers"          slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
+      "class_prefix"      string => canonify(join("_", "promisers"));
+      "args"               slist => { "${variable_prefix}", "${variable_name}", "${file_name}" };
+
+      # define the variable within the variable_prefix namespace
+      "${variable_prefix}.${variable_name}"  data => readdata("${file_name}", "${file_type}");
```
Comment (peckpeck) [inline tree/30_generic_methods/variable_dict_from_file_type.cf]: There is an ordering problem, file_type is defined in pass2, which means:
- the file is read twice
- the output class that are defined at first pass will be wrong

**Comment by jooooooon** [comment]
> Not tested, do not merge for now.

It would be good to have an automatic test for this method (or extend the existing one if applicable) too, so it's not just your tests, but all tests for the whole of ncf's future :)

**Comment by peckpeck** [comment]
ping @amousset

**Comment by peckpeck** [comment]
ping @amousset

**Comment by peckpeck** [comment]
Closed until the agent includes yaml and xml

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
PR replaced by https://github.com/Normation/ncf/pull/998

**Pull request closed without merging**